src/com/griddynamics/cinema/Cinema.java

```diff
@@ -0,0 +1,82 @@
+package com.griddynamics.cinema;
+
+import java.util.stream.Stream;
+
+public class Cinema {
+	
+    final char[][] HALL;
```

Comment:
Think about refactor this `hall` variable, create object for it.  Your business logic depends on representation, it should be opposite (for example, you look at a char - is seat booked or not), which is not good.
My suggestion is to create class Seat, which will contain information about is it reserved or not. And create a HallRow, which will contain list of Seats and row number. You can calculate there reserved seat by each row.